### ISSUE TITLE
consumer.poll() on empty topic returns MAX_POLL_EXCEEDED after max.poll.interval.ms

### ISSUE THREAD

**Issue by danmilon** [confluentinc/confluent-kafka-python#618]
Description
===========

When subscribed to an empty topic, `consumer.poll(timeout=None)` returns, after `max.poll.interval.ms`, a message with `msg.error()`:
```
KafkaError{code=_MAX_POLL_EXCEEDED,val=-147,str="Application maximum poll interval (10000ms) exceeded by 15ms"}
```

Expected Behavior
================

The docs mention that `max.poll.interval.ms` is the maximum interval _between_ `poll()` calls, but clearly in the following script, it raises while _inside_ `poll()`, not between `poll()` calls.
```
Maximum allowed time between calls to consume messages (e.g., rd_kafka_consumer_poll()) for high-level consumers. If this interval is exceeded the consumer is considered failed and the group will rebalance in order to reassign the partitions to another consumer group member.
```

How to reproduce
================

Run the following script by pointing it to an empty topic and wait for 10 seconds:
```python
import sys
import confluent_kafka as kafka

consumer = kafka.Consumer({
    "bootstrap.servers": sys.argv[1],
    "group.id": "group_id",
    "enable.auto.commit": "false",
    "max.poll.interval.ms": "10000",
})

consumer.subscribe([sys.argv[2]])

print("poll: start")
msg = consumer.poll()
print("poll: end")
if msg.error():
    raise msg.error()
print("end")

```
Call it as `python reproduce.py <bootstrap-server> <empty-topic-name>```

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`): 
```
>>> kafka.version()
('1.0.0', 1048576)
>>> kafka.libversion()
('1.0.0', 16777471)
```
 - [x] Apache Kafka broker version: `2.1.0`
 - [ ] Client configuration: `{...}`
 - [x] Operating system: ArchLinux 
 - [ ] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
This is a bug in librdkafka, it currently only records the last poll time at the time the application calls poll(), so it needs to be called at least as often as max.poll.interval.ms.
It'll be fixed in v1.0.0, in the meantime, make sure the poll timeout < max.poll.interval.ms

**Comment by danmilon** [comment]
Oh, that explains it. Thanks for the prompt reply, I'll do that as a workaround.

**Comment by rnpridgeon** [comment]
Fixed in 1.1.0
